src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1029,24 +1031,306 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitState
+    // size. Since each expand step (done in _AdvanceSnappedDimension()) guarantees that all the
+    // sizes will be snapped, our return values is also snapped.
+    //   Why do we do it this, iterative way? Why can't we just split the given size by
+    // _desiredSplitPosition and snap it latter? Because it's hardly doable, if possible, to also
+    // fulfill the monotonicity requirement that way. As the fullSize increases, the proportional
+    // point that separates children panes also moves and cells sneak in the available area in
+    // unpredictable way, regardless which child has the snap priority or whether we snap them
+    // upward, downward or to nearest.
+    //   With present way we run the same sequence of actions regardless to the fullSize value and
+    // only just stop at various moments when the built sizes reaches it.  Eventually, this could
+    // be optimized for simple cases like when both children are both leaves with the same character
+    // size, but it doesn't seem to be beneficial.
+
+    auto sizeTree = _CreateMinSizeTree(widthOrHeight);
+    LayoutSizeNode lastSizeTree{ sizeTree };
+
+    while (sizeTree.size < fullSize)
+    {
+        lastSizeTree = sizeTree;
+        _AdvanceSnappedDimension(widthOrHeight, sizeTree);
+
+        if (sizeTree.size == fullSize)
+        {
+            // If we just hit exactly the requested value, then just return the
+            // current state of children.
+            return { { sizeTree.firstChild->size, sizeTree.secondChild->size },
+                     { sizeTree.firstChild->size, sizeTree.secondChild->size } };
+        }
+    }
+
+    // We exceeded the requested size in the loop above, so lastSizeTree will have
+    // the last good sizes (so that children fit in) and sizeTree has the next possible
+    // snapped sizes. Return them as lower and higher snap possibilities.
+    return { { lastSizeTree.firstChild->size, lastSizeTree.secondChild->size },
+             { sizeTree.firstChild->size, sizeTree.secondChild->size } };
+}
+
+// Method Description:
+// - Adjusts given dimension (width or height) so that all descendant terminals
+//   align with their character grids as close as possible. Snaps to closes match
+//   (either upward or downward). Also makes sure to fit in minimal sizes of the panes.
+// Arguments:
+// - widthOrHeight: if true operates on width, otherwise on height
+// - dimension: a dimension (width or height) to snap
+// Return Value:
+// - A value corresponding to the next closest snap size for this Pane, either upward or downward
+float Pane::CalcSnappedDimension(const bool widthOrHeight, const float dimension) const
+{
+    const auto snapPossibilites = _CalcSnappedDimension(widthOrHeight, dimension);
+    const auto lower = snapPossibilites.lower;
+    const auto higher = snapPossibilites.higher;
+    return dimension - lower < higher - dimension ? lower : higher;

```

Comment:
nit: this could use some parens, it definitely did my eyes in

Reply:
I had it but forgot to commit apparently.

Reply:
This doesn't look resolved...

Reply:
But it is, I guess? And for some reason github doesn't show that? I went into file changes, selected "Changes from all commits" and I see it (now it's 2 lines instead of 4). Unless I have misunderstood the request and that's not what you wanted.

Reply:
Yeah github should mark it as outdated, but I think Dustin might also want something like
`return (dimension - lower) < (higher - dimension) ? lower : higher;`?